dockers/docker-sonic-telemetry/certificate_rotation_checker

```diff
@@ -0,0 +1,205 @@
+        if exit_code != 0:
+            syslog.syslog(syslog.LOG_INFO,
+                          "Failed to restart telemetry server process and retry after 60 seconds ...")
+            time.sleep(60)
+        else:
+            break
+
+    if retry_times > MAX_RETRY_TIMES:
+        syslog.syslog(syslog.LOG_ERR,
+                      "Failed to restart telemetry server process after trying '{}' times and exiting ..."
+                      .format(MAX_RETRY_TIMES))
+        sys.exit(2)
+
+    check_telemetry_server_running()
+
+
+def check_certificate_rotated(certificate_file_name):
+    """Leverages the 'inotify' module to monitor the file system events under the
+    directory which stores the SONiC credentials and restarts telemetry server
+    process if its certificate was rotated.
+
+
+    Args:
+        certificate_file_name: A string indicates the telemetry certificate file name.
+
+    Returns:
+        None.
+    """
+    certificate_file_rotated = False
+
+    inotify_instance = inotify.adapters.Inotify()
+    inotify_instance.add_watch(CREDENTIALS_DIR_PATH)
+    for event in inotify_instance.event_gen(yield_nones=False):
+        header, event_type, monitoring_path, file_name = event
+        if (file_name == certificate_file_name
+                and ("IN_CREATE" in event_type or "IN_MOVED_TO" in event_type)):
+            certificate_file_rotated = True
+
+        if certificate_file_rotated:
+            certificate_file_rotated = False
+            syslog.syslog(syslog.LOG_INFO,
+                          "Certificate was rotated and restarting telemetry server process ...")
+            restart_telemetry_server()
+
+
+def certificate_rotated_checker():
+    """Checks rotation of certificate file and restart streaming telemetry server if necessary.
+
+    Leverages 'inotify' module to check whether the certificate file of streaming telemetry was
+    already rotated or not. The streaming telemetry server process will be restarted if it was rotated.
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
+    certificate_file_path = ""
+    certificate_file_name = ""
+
+    config_db = swsscommon.DBConnector("CONFIG_DB", 0)
+    telemetry_table = swsscommon.Table(config_db, "TELEMETRY")
+    telemetry_table_keys = telemetry_table.getKeys()
+    if "certs" in telemetry_table_keys:
+        certs_info = dict(telemetry_table.get("certs")[1])
+        if "server_crt" in certs_info:
+            certificate_file_path = certs_info["server_crt"]
+            syslog.syslog(syslog.LOG_INFO, "Path of certificate file is '{}'".format(certificate_file_path))
+        else:
+            syslog.syslog(syslog.LOG_ERR,
+                          "Failed to retrieve the path of certificate file from 'TELEMETRY' table!")
+            sys.exit(3)
+    else:
+        syslog.syslog(syslog.LOG_ERR,
+                      "Failed to retrieve the certificate information from 'TELEMETRY' table!")
+        sys.exit(4)
+
+    while True:
```

Comment:
>while True: [](http://example.com/codeflow?start=4&length=11)

The intention of this loop is to wait until two files exist. why only check one file?

Reply:
Updated and checks the existence of both two files.